src/clientdlg.cpp

```diff
@@ -730,15 +722,9 @@ void CClientDlg::OnConnectDlgAccepted()
             }
         }
 
-        // first check if we are already connected, if this is the case we have to
-        // disconnect the old server first
-        if ( pClient->IsRunning() )
-        {
-            Disconnect();
-        }
-
```

Comment:
I don't think this code should be removed. Or else equivalent code should be added somewhere according to the new structure (which I don't yet fully understand).

It is possible, while connected to a server, to go `File`->`Connection Setup...`, or `Ctrl-C`, to bring up the connect dialog, and then select a new server to connect to. In existing versions of Jamulus, this disconnects from the connected server, and then immediately connects to the new server. This was used to great effect during WorldJam to quickly move between rooms.

In the version built from this PR, this doesn't work. The existing connection continues instead of connecting to the new server.

Reply:
We could probably call something like pClient->Disconnect()/Stop() and then pClient->Connect().
The main idea of this PR is to:
1. Emit signals if the client disconnects or connects
2. Introduce convenient methods for programmatic (JSON-RPC) disconnection and connection.

Reply:
Ok. Fixed. Not sure if we need the if is running then stop.
I'd like to have this in the client then.